packages/governance/src/contractGovernorKit.js

```diff
@@ -1,16 +1,57 @@
 import { Fail } from '@agoric/assert';
-import { makeTracer } from '@agoric/internal';
-import { prepareExoClassKit } from '@agoric/vat-data';
+import { UnguardedHelperI, makeTracer } from '@agoric/internal';
+import { M, prepareExoClassKit } from '@agoric/vat-data';
 import { E } from '@endo/eventual-send';
+import {
+  InvitationShape,
+  InstanceHandleShape,
+  InstallationShape,
+} from '@agoric/zoe/src/typeGuards.js';
+import { TimestampShape } from '@agoric/time';
 import { setupApiGovernance } from './contractGovernance/governApi.js';
 import { setupFilterGovernance } from './contractGovernance/governFilter.js';
 import {
   CONTRACT_ELECTORATE,
   setupParamGovernance,
 } from './contractGovernance/governParam.js';
+import { ClosingRuleShape, ParamChangesSpecShape } from './typeGuards.js';
 
 const trace = makeTracer('CGK', false);
 
+const ContractGovernorKitI = {
+  helper: UnguardedHelperI,
+  creator: M.interface('Contract Governor Kit creator', {
+    replaceElectorate: M.call(InvitationShape).returns(M.promise()),
+    voteOnParamChanges: M.call(
+      InstallationShape,
+      TimestampShape,
+      ParamChangesSpecShape,
+    ).returns(M.promise()),
+    voteOnApiInvocation: M.call(
+      M.string(),
+      M.arrayOf(M.any()),
+      InstallationShape,
+      TimestampShape,
+    ).returns(M.promise()),
+    voteOnOfferFilter: M.call(
+      InvitationShape,
+      TimestampShape,
+      M.arrayOf(M.string()),
+    ).returns(M.promise()),
+    getCreatorFacet: M.call().returns(M.remotable('ElectorateCreator')),
+    getAdminFacet: M.call().returns(M.remotable('ElectorateAdmin')),
+    getInstance: M.call().returns(InstanceHandleShape),
+    getPublicFacet: M.call().returns(M.remotable('ElectoratePublic')),
+  }),
+  public: M.interface('Contract Governor Kit public', {
+    getElectorate: M.call().returns(M.promise(/* Instance */)),
```

Comment:
Isn't this also `InstanceHandleShape`?

Reply:
guards can't (yet?) inspect promises

Reply:
It looks to me as if `getElectorate()` and `getGovernedContract()` either both return promises or both return `Instance`s. My guess would be that internal to `contractGovernorKit`, the guard can detect the actual instance, even though the (remote) caller is going to receive a promise, so these can both be declared `InstanceHandleShape`.

Reply:
<img width="598" alt="Screenshot 2023-12-13 at 7 48 28 AM" src="https://github.com/Agoric/agoric-sdk/assets/21505/29e8b29f-ad2a-47d9-b997-0df284fd2f41">
<img width="527" alt="Screenshot 2023-12-13 at 7 48 46 AM" src="https://github.com/Agoric/agoric-sdk/assets/21505/5035e944-ae7a-473b-9918-5a7a7ec8282b">

Reply:
got it.  Thanks.